plugins/source/aws/resources/services/ec2/images.go

```diff
@@ -33,3 +39,58 @@ func Images() *schema.Table {
 		},
 	}
 }
+
+func fetchEc2Images(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	c := meta.(*client.Client)
+
+	svc := c.Services().Ec2
+	g, ctx := errgroup.WithContext(ctx)
+	g.Go(func() error {
+		// fetch ec2.Images owned by this account
+		pag := ec2.NewDescribeImagesPaginator(svc, &ec2.DescribeImagesInput{
+			Owners: []string{"self"},
+		})
+		for pag.HasMorePages() {
+			resp, err := pag.NextPage(ctx)
+			if err != nil {
+				return err
+			}
+			res <- resp.Images
+		}
+		return nil
+	})
+
+	g.Go(func() error {
+		// fetch ec2.Images that are shared with this account
+		pag := ec2.NewDescribeImagesPaginator(svc, &ec2.DescribeImagesInput{
+			ExecutableUsers: []string{"self"},
+		})
+		for pag.HasMorePages() {
+			resp, err := pag.NextPage(ctx)
+			if err != nil {
+				return err
+			}
+			for _, image := range resp.Images {
+				if aws.ToString(image.OwnerId) != c.AccountID {
+					res <- resp.Images
+				}
```

Comment:
Added filtering to exclude any Images that are owned by the same accountId

Reply:
You mean "to exclude any images that are NOT owned by the same account ID", right?

Reply:
Or no... maybe I'm not understanding this properly

Reply:
Oh I see, first we list images we own, then images we can see (and filter out the ones we own)

Reply:
Exactly! There is no way to have multiple filters to get both the AMIs that the account owns as well as the AMIs that are shared with the account, so we have to have multiple calls